core/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -38,6 +38,7 @@
  */
 public class MultiSearchRequest extends ActionRequest<MultiSearchRequest> implements CompositeIndicesRequest {
 
+    private int maxParallelSearchRequests = 10;
```

Comment:
Perhaps the limit should be based on nodes so eg max of N concurrent search requests per node?